### DISCUSSION TITLE
LONG cables -- tacho signals go crazy

### DISCUSSION THREAD

**Discussion by chrismyers81** [tjko/fanpico#76] [Q&A]
So this is definitely outside of the intended use case, but I'm curious if anyone has thoughts.

I'm using my fanpico to control various fans in our basement. Some are within a foot of the controller, and two of them are about 15 feet away (connected with cat5e cable.) That distance works fine.

I'm trying to bring on another two fans with about 30 feet of cat5e cable in between the controller and the fans. When just the cable is hooked up, everything seems fine. However as soon as I plug a fan (any fan, have tried multiples) into the cable, the tacho numbers go crazy on all channels (even ones without fans hooked up.) If I leave the cables plugged into those ports, but use jumpers to connect the fan close to the fanpico, things look ok. So it seems like there's some sort of interference possibly on the long runs? It's weird though that it impacts all of the tacho numbers, not just the one(s) the fan is connected to.

Thinking that maybe it might possibly be interference, etc., but not sure? Any thoughts?

**Comment by tjko** [comment]
I think the long cable could be acting as RF antenna. Using shielded CAT cable (and connecting the shield to ground on one end only) might help?   If you have oscilloscope then hooking it on the tachometer pin (pin 3) might shed some insight what is happening...

How is the CAT5e wired?  Are you using one pair for GND and one pair for 12V power and the remaining two pairs for PWM and Tachometer signals (connecting the second wire on those pairs to GND)?

Currently FanPico does not have TVS diodes to protect inputs/outputs, since it was meant to be installed inside a (computer) case, etc...  (something that probably should be added next PCB revision, at least the footprints so those could be easily added when needed).

It might be worth testing if adding 12v TVS diode between GND (pin 1) and 12V (pin 2), and 5V TVS diodes between Tacho (pin 3) and GND (pin 1) and PWM (pin 4) and GND (pin 1) would make any difference...
